src/components/NewDatePicker/index.js

```diff
@@ -77,13 +84,12 @@ class NewDatePicker extends React.Component {
                 return;
             }
             this.setState({isPickerVisible: false});
-            this.textInputRef.blur();
         });
     }
 
     render() {
         return (
-            <>
+            <View ref={ref => this.wrapperRef = ref} style={styles.datePickerRoot}>
```

Comment:
- Add `onBlur` callback to view to hide the picker if the calendar was focused but not the text input.
- Remove `onBlur` callback from TextInput - It's going to be redundant since events bubble.

Reply:
removing onBlur from text input will prevent closing calendar when pressing outside it